src/VisualStudio/Core/Def/CallHierarchy/CallHierarchyCommandHandler.cs

```diff
@@ -58,53 +63,43 @@ public CallHierarchyCommandHandler(
 
     public bool ExecuteCommand(ViewCallHierarchyCommandArgs args, CommandExecutionContext context)
     {
+        var document = args.SubjectBuffer.CurrentSnapshot.GetOpenDocumentInCurrentContextWithChanges();
+        if (document == null)
+            return false;
+
+        var point = args.TextView.Caret.Position.Point.GetPoint(args.SubjectBuffer, PositionAffinity.Predecessor);
+        if (point is null)
+            return false;
+
         // We're showing our own UI, ensure the editor doesn't show anything itself.
         context.OperationContext.TakeOwnership();
         var token = _listener.BeginAsyncOperation(nameof(ExecuteCommand));
-        ExecuteCommandAsync(args, context)
+        ExecuteCommandAsync(document, point.Value.Position)
             .ReportNonFatalErrorAsync()
             .CompletesAsyncOperation(token);
 
         return true;
     }
 
-    private async Task ExecuteCommandAsync(ViewCallHierarchyCommandArgs args, CommandExecutionContext commandExecutionContext)
+    private async Task ExecuteCommandAsync(Document document, int caretPosition)
     {
-        Document document;
-
         using (var context = _threadOperationExecutor.BeginExecute(
             EditorFeaturesResources.Call_Hierarchy, ServicesVSResources.Navigating, allowCancellation: true, showProgress: false))
         {
-            document = await args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(
-                commandExecutionContext.OperationContext).ConfigureAwait(true);
-            if (document == null)
-            {
-                return;
-            }
-
-            var caretPosition = args.TextView.Caret.Position.BufferPosition.Position;
```

Comment:
note: this was broken.  we need to map the caret pos down to the subject buffer (which i now do above).  otherwise projection scenarios do not work.